### PR TITLE
test/mon: initialize member variable to avoid segfault

### PR DESCRIPTION
**Pull request by ztczll**
…n setup () is called for the first time

Signed-off-by: zhanglei <243290414@qq.com>

### CODE REVIEW

File: src/test/mon/test-mon-msg.cc [hunk 1]
```diff
@@ -219,7 +219,7 @@ class MonMsgTest : public MonClientHelper,
 {
 protected:
   int reply_type;
-  Message *reply_msg;
+  Message *reply_msg = NULL;
```
Comment (jecluis) [inline src/test/mon/test-mon-msg.cc]: initialize it in the constructor instead, please.

**Comment by ztczll** [comment]
I have modified, thank you for your suggestion.

File: src/test/mon/test-mon-msg.cc [hunk 2]
```diff
@@ -224,6 +224,7 @@ class MonMsgTest : public MonClientHelper,
   Cond cond;
 
   MonMsgTest() :
+    reply_msg(NULL),
```
Comment (tchaikov) [inline src/test/mon/test-mon-msg.cc]: could also initialize it using in-class initialization, see http://en.cppreference.com/w/cpp/language/data_members

File: src/test/mon/test-mon-msg.cc (hunk 1 above)
Comment (tchaikov) [inline src/test/mon/test-mon-msg.cc]: @jecluis hold on, why?

**Comment by ztczll** [comment]
Does this still need to be modified？

**Comment by tchaikov** [comment]
@ztczll i am fine either way. just don't understand why @jecluis does not like 
```c++
Message *reply_msg = NULL;
```

@jecluis does the latest change address your comment?

**Pull request merged**